esp-hal/src/i2c/master/mod.rs

```diff
@@ -660,8 +667,8 @@ struct I2cFuture<'a> {
 
 #[cfg(not(esp32))]
 impl<'a> I2cFuture<'a> {
-    pub fn new(event: Event, instance: &'a impl Instance) -> Self {
-        instance.info().register_block().int_ena().modify(|_, w| {
+    pub fn new(event: Event, info: &'a Info, state: &'a State) -> Self {
```

Comment:
Any reason not to take a `Driver` here? Instead of info and state.

Reply:
None, it's part of the indecision I have with this PR.